hc/front/views.py

```diff
@@ -27,6 +32,112 @@ def pairwise(iterable):
+                name = category_form.cleaned_data['category']
+                ctg = Category(name = name)
+                ctg.save()
+                return redirect(create_blog)
+        elif 'create_blog' in request.POST:
+            if form.is_valid():
+                title = request.POST['title'] 
+                blog = form.cleaned_data['content']
+                selected_category = request.POST['category_name']
+                category = Category.objects.get(name=selected_category)
+                published = timezone.now()
+                user = request.user
+                blog = Blog_post(title=title, content=blog, category=category,
+                                published=published, user=user)
+                blog.save()
+                messages.add_message(request, messages.INFO, 'Successfully created blog')
+                return redirect(read_blog, pk=blog.id)
+    else:
+        categories = Category.objects.all()
+        ctx = {
+            'category':categories,
+            'form':CreateBlogPost({'content':'Write Blog'}),
+            'category_form':category_form,
+            'edit': False
+            }
+        return render(request, 'front/create_blog.html', ctx)
+
+
+def read_blog(request, pk):
+    comment_form = CreateCommentForm(request.POST)
+    blog = Blog_post.objects.get(pk=pk)
+    featured = Blog_post.objects.get(pk=pk)
+    comments = Comment.objects.filter(blog = blog.id)
+    url = str.format("http://localhost:8000/blog/read_blog/{pk}")
+    ctx = {
+        'blog': blog,
+        'featured':featured,
+        'tweet_url':url,
+        'comments':comments
+    }
+    if 'add_comment' in request.POST:
+        if comment_form.is_valid():
+            posted_comment = request.POST['comment']
+            published = timezone.now()
+            comment = Comment(comment = posted_comment, blog = blog, user = request.user, published=published)
+            comment.save()
+            return redirect(read_blog, pk=blog.id)
+    return render(request, 'front/readblog.html', ctx )
+
+
+def delete_blog(request, pk):
+    deleted_blog = Blog_post.objects.get(pk=pk)
+    Blog_post.objects.get(pk=pk).delete()
+    messages.add_message(request, messages.INFO, 'Successfully deleted blog')
+    return redirect(blogs, filter_by=0)
+
+
+def edit_blog(request, pk):
+    blog =  Blog_post.objects.get(pk=pk)
+    Category = blog.category
+    if request.method == 'POST':
+        form = CreateBlogPost(request.POST)
+        category_form = CreateCategory(request.POST)
+        if "create_blog" in request.POST:
+            if form.is_valid():
+                blog.category = blog.category
+                blog.title = request.POST['title']
+                blog.content = request.POST['content']
+                blog.publish = timezone.now()
+                blog.save()
+                messages.add_message(request, messages.INFO, 'Successfully edited blog')
+                return redirect(read_blog, pk)
```

Comment:
So many nested `if`s. strive not to have more than 2.
conjoin `form.is_valid()` with the outer conditional statement.

Reply:
Noted and fixed